unlock-app/src/components/interface/keychain/TransferModal.tsx

```diff
@@ -0,0 +1,253 @@
+import { useQuery } from '@tanstack/react-query'
+import {
+  AddressInput,
+  Button,
+  Input,
+  Modal,
+  Placeholder,
+} from '@unlock-protocol/ui'
+import { useState } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { MAX_UINT } from '~/constants'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { onResolveName } from '~/utils/resolvers'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+
+interface TransferModalProps {
+  isOpen: boolean
+  setIsOpen: (state: boolean) => void
+  network: number
+  owner: string
+  tokenId: string
+  expiration: any
+  lock: any
+}
+
+interface TransferFormProps {
+  recipientAddress: string
+  time: number
+}
+
+type TransferOption = 'transfer' | 'share' | 'lend' | undefined
+
+const MAX_TRANSFER_FEE = 10000 //100% of transfer fees = disabled transfers
+
+export function TransferModal({
+  isOpen,
+  setIsOpen,
+  lock,
+  network,
+  owner,
+  tokenId,
+  expiration,
+}: TransferModalProps) {
+  const hasNoExpiry = expiration === MAX_UINT
+
+  const { getWalletService } = useAuth()
+  const web3service = useWeb3Service()
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [option, setOption] = useState<TransferOption>(undefined)
+
+  const maxTransferableSeconds = expiration - Math.floor(Date.now() / 1000)
+  const maxTransferableDays = Math.floor(
+    maxTransferableSeconds / (60 * 60 * 24)
+  )
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<TransferFormProps>({
+    mode: 'onChange',
+    defaultValues: {
+      recipientAddress: undefined,
+      time: undefined,
+    },
+  })
+
+  async function fetchTransferFees() {
+    let result = await web3service.transferFeeBasisPoints(lock.address, network)
+    result = Number(result)
+    return result
+  }
+
+  const {
+    isLoading: pageIsLoading,
+    error,
+    data: transferFees,
+  } = useQuery({
```

Comment:
Please do that inside of a dedicated hook like `useLockTransferFees` (like the rest of the app does!)